app/forms/steps/client/contact_details_form.rb

```diff
@@ -8,14 +8,26 @@ class ContactDetailsForm < Steps::BaseFormObject
 
       attribute :telephone_number, :string
 
+      attribute :correspondence_address_type, :correspondence_address_type
+
       has_one_association :applicant
 
-      validates :telephone_number, format: { with: TEL_REGEXP }, presence: true
+      validates :telephone_number,
+                format: { with: TEL_REGEXP },
+                presence: true
+
+      validates :correspondence_address_type,
+                inclusion: { in: :choices },
+                presence: true
 
       def telephone_number=(str)
         super(str.delete(' ')) if str
       end
 
+      def choices
```

Comment:
Oh I just recalled, the choices are dynamic, as in, if the provider didn't enter a home address, then the Home address choice will not show.
But that can be done in a separate small PR if you prefer 👍

Reply:
I'll raise in a separate PR 👍